src/Configuration/RectorConfigBuilder.php

```diff
@@ -165,10 +166,22 @@ final class RectorConfigBuilder
 
     private ?bool $isWithPhpLevelUsed = null;
 
+    /**
+     * @var array<class-string<SetProviderInterface>,bool>
+     */
+    private array $setProviders = [];
+
     public function __invoke(RectorConfig $rectorConfig): void
     {
-        if ($this->setGroups !== []) {
-            $setManager = new SetManager(new SetProviderCollector(), new InstalledPackageResolver(getcwd()));
+        if ($this->setGroups !== [] || $this->setProviders !== []) {
+            $setProviderCollector = new SetProviderCollector(array_map(
+                static fn (string $setProvider): SetProviderInterface =>
+                $rectorConfig->make($setProvider),
```

Comment:
can this be a new instance creation over use service, eg;

```suggestion
               new $setProvider(),
```

just to make consistent with the way usage of new on 

https://github.com/rectorphp/rector-src/blob/fe8e48de8cc6c962430abab1b1766c05dd12309e/src/Bridge/SetProviderCollector.php#L36-L41

Reply:
I used this syntax so that the "Dependency Injection Container" can be used to "autowire" dependencies for a user created "set provider".

based on your extensive knowledge of this codebase, do you have a suggestion on how to address both?

- new instance creation
- autowiring dependencies

Reply:
ok, make sense 👍